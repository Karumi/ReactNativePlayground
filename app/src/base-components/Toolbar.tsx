import {  Body, Header, Title } from "native-base";
import * as React from "react";
import translator from "../i18n";

class Toolbar extends React.Component {
    public render() {
        return (
            <Header>
                <Body>
                    <Title>{translator().appName}</Title>
                </Body>
            </Header>
        );
    }
}

export default Toolbar;
