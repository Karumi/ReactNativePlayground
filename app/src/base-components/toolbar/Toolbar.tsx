import { Body, Button, Header, Left, Title } from "native-base";
import * as React from "react";
import { Image } from "react-native";
import translator from "../../i18n";
import { ios } from "../../platform";
import assets from "./images/assets";

export interface Props {
    navigation: any;
    showBackButton?: boolean;
    title?: string;
}

class Toolbar extends React.Component<Props> {

    public static defaultProps: Partial<Props> = {
        showBackButton: false,
        title: translator().appName,
    };

    public render() {
        const showBack = this.props.showBackButton;
        const titleLeftPadding = ios() && showBack ? -12 : 0;
        const leftItem = showBack ? this.backButton() : null;
        return (
            <Header>
                <Left style={{ flex: 0, paddingLeft: 0 }} >
                    {leftItem}
                </Left>
                <Body style={{ left: titleLeftPadding }}>
                    <Title>{this.props.title}</Title>
                </Body>
            </Header>
        );
    }
    private backButton() {
        return (
            <Button transparent onPress={() => this.props.navigation.pop()}>
                <Image source={assets.arrowBack} />
            </Button>
        );
    }
}

export default Toolbar;
