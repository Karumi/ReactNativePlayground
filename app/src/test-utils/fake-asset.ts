jest.mock("../assets/asset", () => {
return {
        default: (uri) => uri,
    };
});
