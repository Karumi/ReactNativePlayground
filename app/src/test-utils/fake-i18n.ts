
import { I18nResolver } from "i18n-ts";
import en from "../translations/en";

const i18n = { default: en };

jest.mock("../i18n", () => {
return {
        default: () => new I18nResolver(i18n,  "en").translation,
    };
} );
