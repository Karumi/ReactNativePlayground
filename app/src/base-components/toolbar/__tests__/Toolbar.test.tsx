
import { Button } from "native-base";
import * as React from "react";
import * as renderer from "react-test-renderer";
import "../../../test-utils/fake-i18n";
import Toolbar from "../Toolbar";

jest.mock("../images/assets", () => {
    return {
        default: {
            arrowBack: "arrowBack",
        },
    };
});

it("renders correctly with defaults", () => {
    const toolbar = renderer.create(<Toolbar />).toJSON();
    expect(toolbar).toMatchSnapshot();
});

it("renders in back button mode if the showBackButton property is true", () => {
    const toolbar = renderer.create(<Toolbar showBackButton={true} />).toJSON();
    expect(toolbar).toMatchSnapshot();
});

it("overrides the default app title if the title prop is configured", () => {
    const toolbar = renderer.create(<Toolbar title="TEST TITLE" />).toJSON();
    expect(toolbar).toMatchSnapshot();
});

it("navigates back if the back button is tapped", () => {
    const mockNavigation = {
        pop: jest.fn(),
    };
    const toolbar = renderer.create(<Toolbar showBackButton={true} navigation={mockNavigation} />).root;

    const backButton = toolbar.findByType(Button).instance;
    backButton.props.onPress();

    expect(mockNavigation.pop).toHaveBeenCalled();
});
