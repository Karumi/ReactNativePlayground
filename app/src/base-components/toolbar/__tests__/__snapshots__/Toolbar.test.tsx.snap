// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overrides the default app title if the title prop is configured 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderBottomColor": "#a7a6ab",
        "borderBottomWidth": 0.5,
        "elevation": 3,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "center",
        "left": 0,
        "paddingLeft": 6,
        "paddingRight": 10,
        "paddingTop": 18,
        "right": 0,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
        "top": 0,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          },
          Object {
            "flex": 0,
            "paddingLeft": 0,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          },
          Object {
            "left": 0,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 17,
            "fontWeight": "700",
            "marginLeft": undefined,
            "paddingLeft": 4,
            "paddingTop": 1,
            "textAlign": "center",
          }
        }
      >
        TEST TITLE
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly with defaults 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderBottomColor": "#a7a6ab",
        "borderBottomWidth": 0.5,
        "elevation": 3,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "center",
        "left": 0,
        "paddingLeft": 6,
        "paddingRight": 10,
        "paddingTop": 18,
        "right": 0,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
        "top": 0,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          },
          Object {
            "flex": 0,
            "paddingLeft": 0,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          },
          Object {
            "left": 0,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 17,
            "fontWeight": "700",
            "marginLeft": undefined,
            "paddingLeft": 4,
            "paddingTop": 1,
            "textAlign": "center",
          }
        }
      >
        React Native Playground EN
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders in back button mode if the showBackButton property is true 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderBottomColor": "#a7a6ab",
        "borderBottomWidth": 0.5,
        "elevation": 3,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "center",
        "left": 0,
        "paddingLeft": 6,
        "paddingRight": 10,
        "paddingTop": 18,
        "right": 0,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
        "top": 0,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          },
          Object {
            "flex": 0,
            "paddingLeft": 0,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": null,
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "marginLeft": -3,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 4,
            "paddingRight": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Image
          source="arrowBack"
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          },
          Object {
            "left": -12,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 17,
            "fontWeight": "700",
            "marginLeft": undefined,
            "paddingLeft": 4,
            "paddingTop": 1,
            "textAlign": "center",
          }
        }
      >
        React Native Playground EN
      </Text>
    </View>
  </View>
</View>
`;
